### ISSUE TITLE
[bug]: failed to run script file: ods-jenkins-shared-library.groovy

### ISSUE THREAD

**Issue by SimonGolms** [opendevstack/ods-core#316]
## Describe the bug
Jenkins is throwing a warning during the deployment process. 

### OpenShift

1. Go to `hugo-cd`
2. Trigger an new deployment of jenkins 
3. Go to the `Logs` tab of running jenkins pod
4. See warning

## Expected behavior

Deployment without warnings / errors / stack trace

## Screenshots

![image](https://user-images.githubusercontent.com/17799657/70242483-34a20c80-1771-11ea-95cf-29892f9bfb5d.png)

## Affected version

- **OpenShift:** 3.11
- **OpenDevStack:** 1.2

## Log Output

```
[...]
Dec 05, 2019 1:36:10 PM org.springframework.beans.factory.support.DefaultListableBeanFactory preInstantiateSingletons
INFO: Pre-instantiating singletons in org.springframework.beans.factory.support.DefaultListableBeanFactory@44c515c2: defining beans [filter,legacy]; root of factory hierarchy
INFO: Found existing libraries: []
DEBUG: Adding ODS build shared lib
Dec 05, 2019 1:36:10 PM jenkins.util.groovy.GroovyHookScript execute
WARNING: Failed to run script file:/var/lib/jenkins/init.groovy.d/ods-jenkins-shared-library.groovy
--
  | java.lang.UnsupportedOperationException
  | at java.util.AbstractList.add(AbstractList.java:148)
  | at java.util.AbstractList.add(AbstractList.java:108)
  | at java_util_List$add$0.call(Unknown Source)
  | at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
  | at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
  | at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
  | at ods-jenkins-shared-library.run(ods-jenkins-shared-library.groovy:70)
  | at groovy.lang.GroovyShell.evaluate(GroovyShell.java:585)
  | at jenkins.util.groovy.GroovyHookScript.execute(GroovyHookScript.java:136)
  | at jenkins.util.groovy.GroovyHookScript.execute(GroovyHookScript.java:127)
  | at jenkins.util.groovy.GroovyHookScript.run(GroovyHookScript.java:110)
  | at hudson.init.impl.GroovyInitScript.init(GroovyInitScript.java:41)
  | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  | at java.lang.reflect.Method.invoke(Method.java:498)
  | at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:104)
  | at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:175)
  | at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
  | at jenkins.model.Jenkins$5.runTask(Jenkins.java:1083)
  | at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
  | at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  | at java.lang.Thread.run(Thread.java:748)

[...]
```

**Comment by michaelsauter** [comment]
Hmm. Is this reproducible for you?

I have tried the steps you create with a Jenkins master instance based on master and cannot reproduce your issue. The script is executed without error.

Actually, in your log output, it seems that the second script (mro) is being executed properly. Can you confirm this?

**Comment by clemensutschig** [comment]
how on earth can this abstract list be returned?!

https://github.com/jenkinsci/workflow-cps-global-lib-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/libs/GlobalLibraries.java#L60

list defined @
https://github.com/jenkinsci/workflow-cps-global-lib-plugin/blob/master/src/main/java/org/jenkinsci/plugins/workflow/libs/GlobalLibraries.java#L54

thoughts?

**Comment by tbugfinder** [comment]
Is that test helpful?

https://github.com/jenkinsci/workflow-cps-global-lib-plugin/blob/95ebc4ceceb510cce87f40148a4c69798217936b/src/test/java/org/jenkinsci/plugins/workflow/libs/LibraryStepTest.java#L102

**Comment by clemensutschig** [comment]
a fix I could quickly think of  ...
``` 
if (existingLibs.size == 0 && !existingLibs instanceof ArrayList) {
  existingLibs = new ArrayList<LibraryConfiguration>();
  lobalLibraries.get().setLibraries(existingLibs)
}
```
thoughts anyone?

**Comment by clemensutschig** [comment]
closed by #597
